### ISSUE TITLE
Complex: error: expected identifier before numeric constant

### ISSUE THREAD

**Issue by mgood7123** [mosra/magnum#460]
```
cmake_minimum_required(VERSION 3.4.1)

project(GLIS)

add_subdirectory(dependancies/WINAPI)

set(BUILD_SHARED_LIBS ON)

add_subdirectory(dependancies/freetype-2.10.2)
include_directories(dependancies/freetype-2.10.2/include)

add_subdirectory(dependancies/glm)
include_directories(dependancies/glm)

add_subdirectory(dependancies/corrade) # build with everything
#add_subdirectory(dependancies/corrade EXCLUDE_FROM_ALL) # so only things you use are built

# https://doc.magnum.graphics/magnum/requires-gl.html

set(WITH_GL ON CACHE BOOL "" FORCE)
set(TARGET_GLES ON CACHE BOOL "" FORCE)
set(WITH_EGLCONTEXT ON CACHE BOOL "" FORCE)
#set(TARGET_DESKTOP_GLES ON CACHE BOOL "" FORCE)
#set(TARGET_VK ON CACHE BOOL "" FORCE)
#set(WITH_MAGNUMFONT ON CACHE BOOL "" FORCE)
#set(WITH_MAGNUMFONTCONVERTER ON CACHE BOOL "" FORCE)
add_subdirectory(dependancies/magnum) # build with everything
#add_subdirectory(dependancies/magnum EXCLUDE_FROM_ALL)  # so only things you use are built

include_directories(dependancies)
include_directories(include)

set(cwd ${CMAKE_CURRENT_SOURCE_DIR})
set(GLIS_INCLUDE_DIRECTORIES ${cwd}/include ${cwd}/dependancies PARENT_SCOPE)

add_subdirectory(testBuilder)

testBuilder_add_source(GLIS src/glis/backup/backup.cpp)
testBuilder_add_source(GLIS src/glis/backup/types/framebuffer.cpp)
testBuilder_add_source(GLIS src/glis/backup/types/renderbuffer.cpp)
testBuilder_add_source(GLIS src/glis/backup/types/texture.cpp)
testBuilder_add_source(GLIS src/glis/backup/types/program.cpp)
testBuilder_add_source(GLIS src/glis/compositor/compositor.cpp)
testBuilder_add_source(GLIS src/glis/font/font.cpp)
testBuilder_add_source(GLIS src/glis/internal/fps.cpp)
testBuilder_add_source(GLIS src/glis/internal/internal.cpp)
testBuilder_add_source(GLIS src/glis/internal/log.cpp)
testBuilder_add_source(GLIS src/glis/ipc/ashmem.cpp)
testBuilder_add_source(GLIS src/glis/ipc/ipc.cpp)
testBuilder_add_source(GLIS src/glis/ipc/serializer.cpp)
testBuilder_add_source(GLIS src/glis/ipc/server_core.cpp)
testBuilder_add_source(GLIS src/glis/ipc/shm.cpp)

if (ANDROID)
    testBuilder_add_library(GLIS GLESv3)
    testBuilder_add_library(GLIS android)
    testBuilder_add_library(GLIS log)
elseif(UNIX)
    testBuilder_add_source(GLIS src/glis/internal/xdg-shell-protocol.c)
    testBuilder_add_library(GLIS GLESv2)
    testBuilder_add_library(GLIS pthread)
    testBuilder_add_library(GLIS X11)
    testBuilder_add_library(GLIS wayland-client)
    testBuilder_add_library(GLIS wayland-egl)
endif()

testBuilder_add_library(GLIS Magnum::Magnum)
testBuilder_add_library(GLIS Magnum::GL)
testBuilder_add_library(GLIS freetype)
testBuilder_add_library(GLIS glm)
testBuilder_add_library(GLIS EGL)
testBuilder_add_library(GLIS WinKernel)
testBuilder_build_shared_library(GLIS)

# BUILD PROGRAMS

macro(build_program directory name)
    testBuilder_add_library(${name} GLIS)
    testBuilder_add_source(${name} ${directory}/${name}.cpp)
    if(ANDROID)
        testBuilder_build(${name} ${CMAKE_SOURCE_DIR}/executables/Arch/${CMAKE_ANDROID_ARCH_ABI})
    elseif(UNIX)
        testBuilder_build(${name} EXECUTABLES)
    endif()
endmacro()

build_program(examples compositor)
build_program(examples X11)
build_program(examples X11_2)
build_program(examples X11_3)
build_program(examples wayland)
build_program(examples X11_magnum)

build_program(compositor_clients MovingWindows)
build_program(compositor_clients MovingWindowsB)
build_program(compositor_clients compositor_start_drawing)
build_program(compositor_clients compositor_stop_drawing)
build_program(compositor_clients DefaultFramebuffer)
build_program(compositor_clients OffScreenFramebuffer)
build_program(compositor_clients Texture)
build_program(compositor_clients MYPRIVATEAPP)
build_program(compositor_clients shm)
```

```
[ 54%] Building CXX object CMakeFiles/testBuilder_executables_X11_magnum.dir/examples/X11_magnum.cpp.o
In file included from /usr/include/X11/Xlib.h:44,
                 from /usr/include/EGL/eglplatform.h:128,
                 from /usr/include/EGL/egl.h:39,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/common.hpp:18,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/internal.hpp:9,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:5,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:5:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Math/Math.h:45:23: error: expected identifier before numeric constant
   45 | template<class> class Complex;
      |                       ^~~~~~~
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Math/Math.h:45:23: error: expected unqualified-id before numeric constant
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Magnum.h:35,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/GL/Context.h:40,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Math/Math.h:97:68: error: template argument 1 is invalid
   97 | template<class T> using CubicHermiteComplex = CubicHermite<Complex<T>>;
      |                                                                    ^
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/GL/Context.h:40,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Magnum.h:773:15: error: ‘CubicHermiteComplex’ in namespace ‘Magnum::Math’ does not name a template type; did you mean ‘CubicHermite3D’?
  773 | typedef Math::CubicHermiteComplex<Float> CubicHermiteComplex;
      |               ^~~~~~~~~~~~~~~~~~~
      |               CubicHermite3D
In file included from /usr/include/X11/Xlib.h:44,
                 from /usr/include/EGL/eglplatform.h:128,
                 from /usr/include/EGL/egl.h:39,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/common.hpp:18,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/internal.hpp:9,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:5,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:5:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Magnum.h:779:15: error: expected unqualified-id before numeric constant
  779 | typedef Math::Complex<Float> Complex;
      |               ^~~~~~~
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/GL/Context.h:40,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Magnum.h:1087:15: error: ‘CubicHermiteComplex’ in namespace ‘Magnum::Math’ does not name a template type; did you mean ‘CubicHermite3D’?
 1087 | typedef Math::CubicHermiteComplex<Double> CubicHermiteComplexd;
      |               ^~~~~~~~~~~~~~~~~~~
      |               CubicHermite3D
In file included from /usr/include/X11/Xlib.h:44,
                 from /usr/include/EGL/eglplatform.h:128,
                 from /usr/include/EGL/egl.h:39,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/common.hpp:18,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/internal/internal.hpp:9,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:5,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:5:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/magnum/src/Magnum/Magnum.h:1093:15: error: expected unqualified-id before numeric constant
 1093 | typedef Math::Complex<Double> Complexd;
      |               ^~~~~~~
make[3]: *** [CMakeFiles/testBuilder_executables_X11_magnum.dir/build.make:63: CMakeFiles/testBuilder_executables_X11_magnum.dir/examples/X11_magnum.cpp.o] Error 1
make[3]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make[2]: *** [CMakeFiles/Makefile2:1131: CMakeFiles/testBuilder_executables_X11_magnum.dir/all] Error 2
make[2]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make[1]: *** [Makefile:152: all] Error 2
make[1]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make: *** [Makefile:26: build_debug] Error 1
```

**Comment by mosra** [comment]
Oh, these errors are the best. :fire: 

There's `#define Complex` in [one of the X11 headers](https://code.woboq.org/qt5/include/X11/X.h.html#_M/Complex), which is the stupidest thing ever, but what can we do. It's being undef'd in Magnum headers that include X11 but I can't really prevent this from happening everywhere, unfortunately...

If you `#undef Complex` after including anything related to X11, EGL or GLX, that should make it work.

**Comment by mgood7123** [comment]
hmmm ok

**Comment by mgood7123** [comment]
if i include Magnum before my glis header (instead of after) i get even more errors...

```
[ 54%] Building CXX object CMakeFiles/testBuilder_executables_X11_magnum.dir/examples/X11_magnum.cpp.o
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/font/font.hpp:10,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:4,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/usr/include/GLES3/gl32.h:1067:9: error: typedef ‘GLsync’ is initialized (use decltype instead)
 1067 | typedef GLsync (GL_APIENTRYP PFNGLFENCESYNCPROC) (GLenum condition, GLbitfield flags);
      |         ^~~~~~
/usr/include/GLES3/gl32.h:1067:30: error: ‘PFNGLFENCESYNCPROC’ was not declared in this scope
 1067 | typedef GLsync (GL_APIENTRYP PFNGLFENCESYNCPROC) (GLenum condition, GLbitfield flags);
      |                              ^~~~~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1068:33: error: typedef ‘PFNGLISSYNCPROC’ is initialized (use decltype instead)
 1068 | typedef GLboolean (GL_APIENTRYP PFNGLISSYNCPROC) (GLsync sync);
      |                                 ^~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1068:51: error: ‘GLsync’ was not declared in this scope
 1068 | typedef GLboolean (GL_APIENTRYP PFNGLISSYNCPROC) (GLsync sync);
      |                                                   ^~~~~~
/usr/include/GLES3/gl32.h:1069:28: error: typedef ‘PFNGLDELETESYNCPROC’ is initialized (use decltype instead)
 1069 | typedef void (GL_APIENTRYP PFNGLDELETESYNCPROC) (GLsync sync);
      |                            ^~~~~~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1069:50: error: ‘GLsync’ was not declared in this scope
 1069 | typedef void (GL_APIENTRYP PFNGLDELETESYNCPROC) (GLsync sync);
      |                                                  ^~~~~~
/usr/include/GLES3/gl32.h:1070:30: error: typedef ‘PFNGLCLIENTWAITSYNCPROC’ is initialized (use decltype instead)
 1070 | typedef GLenum (GL_APIENTRYP PFNGLCLIENTWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1070:56: error: ‘GLsync’ was not declared in this scope
 1070 | typedef GLenum (GL_APIENTRYP PFNGLCLIENTWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                        ^~~~~~
/usr/include/GLES3/gl32.h:1070:80: error: expected primary-expression before ‘flags’
 1070 | IENTRYP PFNGLCLIENTWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1070:96: error: expected primary-expression before ‘timeout’
 1070 |  PFNGLCLIENTWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                                    ^~~~~~~

/usr/include/GLES3/gl32.h:1071:28: error: typedef ‘PFNGLWAITSYNCPROC’ is initialized (use decltype instead)
 1071 | typedef void (GL_APIENTRYP PFNGLWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                            ^~~~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1071:48: error: ‘GLsync’ was not declared in this scope
 1071 | typedef void (GL_APIENTRYP PFNGLWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                ^~~~~~
/usr/include/GLES3/gl32.h:1071:72: error: expected primary-expression before ‘flags’
 1071 | (GL_APIENTRYP PFNGLWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1071:88: error: expected primary-expression before ‘timeout’
 1071 | ENTRYP PFNGLWAITSYNCPROC) (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                                    ^~~~~~~

/usr/include/GLES3/gl32.h:1073:28: error: typedef ‘PFNGLGETSYNCIVPROC’ is initialized (use decltype instead)
 1073 | typedef void (GL_APIENTRYP PFNGLGETSYNCIVPROC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                            ^~~~~~~~~~~~~~~~~~
/usr/include/GLES3/gl32.h:1073:49: error: ‘GLsync’ was not declared in this scope
 1073 | typedef void (GL_APIENTRYP PFNGLGETSYNCIVPROC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                 ^~~~~~
/usr/include/GLES3/gl32.h:1073:69: error: expected primary-expression before ‘pname’
 1073 | id (GL_APIENTRYP PFNGLGETSYNCIVPROC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1073:84: error: expected primary-expression before ‘bufSize’
 1073 | P PFNGLGETSYNCIVPROC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~~~

/usr/include/GLES3/gl32.h:1073:101: error: expected primary-expression before ‘*’ token
 1073 | ROC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^

/usr/include/GLES3/gl32.h:1073:102: error: ‘length’ was not declared in this scope; did you mean ‘glm::length’?
 1073 | OC) (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~~
      |                                                           glm::length
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/../ext/quaternion_geometric.hpp:70,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/quaternion_common.hpp:25,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/quaternion.hpp:20,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/type_ptr.hpp:37,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/font/font.hpp:9,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:4,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/../ext/quaternion_geometric.inl:11:23: note: ‘glm::length’ declared here
   11 |  GLM_FUNC_QUALIFIER T length(qua<T, Q> const& q)
      |                       ^~~~~~
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/font/font.hpp:10,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:4,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/usr/include/GLES3/gl32.h:1073:116: error: expected primary-expression before ‘*’ token
 1073 | GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                                    ^

/usr/include/GLES3/gl32.h:1073:117: error: ‘values’ was not declared in this scope
 1073 | GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                                     ^~~~~~

/usr/include/GLES3/gl32.h:1172:12: error: ‘GLsync’ does not name a type
 1172 | GL_APICALL GLsync GL_APIENTRY glFenceSync (GLenum condition, GLbitfield flags);
      |            ^~~~~~
/usr/include/GLES3/gl32.h:1173:44: error: ‘GLsync’ was not declared in this scope
 1173 | GL_APICALL GLboolean GL_APIENTRY glIsSync (GLsync sync);
      |                                            ^~~~~~
/usr/include/GLES3/gl32.h:1174:43: error: variable or field ‘glDeleteSync’ declared void
 1174 | GL_APICALL void GL_APIENTRY glDeleteSync (GLsync sync);
      |                                           ^~~~~~
/usr/include/GLES3/gl32.h:1174:43: error: ‘GLsync’ was not declared in this scope
/usr/include/GLES3/gl32.h:1175:49: error: ‘GLsync’ was not declared in this scope
 1175 | GL_APICALL GLenum GL_APIENTRY glClientWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                 ^~~~~~
/usr/include/GLES3/gl32.h:1175:73: error: expected primary-expression before ‘flags’
 1175 | num GL_APIENTRY glClientWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1175:89: error: expected primary-expression before ‘timeout’
 1175 | APIENTRY glClientWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                                    ^~~~~~~

/usr/include/GLES3/gl32.h:1175:96: error: expression list treated as compound expression in initializer [-fpermissive]
 1175 | tWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                           ^

/usr/include/GLES3/gl32.h:1176:41: error: variable or field ‘glWaitSync’ declared void
 1176 | GL_APICALL void GL_APIENTRY glWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                         ^~~~~~
/usr/include/GLES3/gl32.h:1176:41: error: ‘GLsync’ was not declared in this scope
/usr/include/GLES3/gl32.h:1176:65: error: expected primary-expression before ‘flags’
 1176 | CALL void GL_APIENTRY glWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1176:81: error: expected primary-expression before ‘timeout’
 1176 | id GL_APIENTRY glWaitSync (GLsync sync, GLbitfield flags, GLuint64 timeout);
      |                                                                    ^~~~~~~

/usr/include/GLES3/gl32.h:1178:42: error: variable or field ‘glGetSynciv’ declared void
 1178 | GL_APICALL void GL_APIENTRY glGetSynciv (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                          ^~~~~~
/usr/include/GLES3/gl32.h:1178:42: error: ‘GLsync’ was not declared in this scope
/usr/include/GLES3/gl32.h:1178:62: error: expected primary-expression before ‘pname’
 1178 | APICALL void GL_APIENTRY glGetSynciv (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~

/usr/include/GLES3/gl32.h:1178:77: error: expected primary-expression before ‘bufSize’
 1178 | _APIENTRY glGetSynciv (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~~~

/usr/include/GLES3/gl32.h:1178:94: error: expected primary-expression before ‘*’ token
 1178 | nciv (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^

/usr/include/GLES3/gl32.h:1178:95: error: ‘length’ was not declared in this scope; did you mean ‘glm::length’?
 1178 | civ (GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                           ^~~~~~
      |                                                           glm::length
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/../ext/quaternion_geometric.hpp:70,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/quaternion_common.hpp:25,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/quaternion.hpp:20,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/type_ptr.hpp:37,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/font/font.hpp:9,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:4,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/dependancies/glm/glm/gtc/../gtc/../ext/../ext/quaternion_geometric.inl:11:23: note: ‘glm::length’ declared here
   11 |  GLM_FUNC_QUALIFIER T length(qua<T, Q> const& q)
      |                       ^~~~~~
In file included from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/font/font.hpp:10,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/include/glis/glis.hpp:4,
                 from /home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:6:
/usr/include/GLES3/gl32.h:1178:109: error: expected primary-expression before ‘*’ token
 1178 | GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                                    ^

/usr/include/GLES3/gl32.h:1178:110: error: ‘values’ was not declared in this scope
 1178 | GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
      |                                                                     ^~~~~~

make[3]: *** [CMakeFiles/testBuilder_executables_X11_magnum.dir/build.make:63: CMakeFiles/testBuilder_executables_X11_magnum.dir/examples/X11_magnum.cpp.o] Error 1
make[3]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make[2]: *** [CMakeFiles/Makefile2:1131: CMakeFiles/testBuilder_executables_X11_magnum.dir/all] Error 2
make[2]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make[1]: *** [Makefile:152: all] Error 2
make[1]: Leaving directory '/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/debug_BUILD'
make: *** [Makefile:26: build_debug] Error 1
smallville7123@smallville7123-MacBookPro:~/AndroidCompositor/app/src/main/jni/GLIS$ 
```

**Comment by mgood7123** [comment]
and if i remove the header then i only get Magnum undeclared

```
/home/smallville7123/AndroidCompositor/app/src/main/jni/GLIS/examples/X11_magnum.cpp:27:14: error: ‘Magnum’ has not been declared
   27 |     auto x = Magnum::GL::Context::current().rendererString();
      |              ^~~~~~
```

**Comment by mgood7123** [comment]
hmmm it appears that i need to include Magnum before egl but after opengl32

```
#include <GLES3/gl32.h>
#include <Magnum/GL/Context.h>
#include <EGL/egl.h> // includes X11 headers on unix
```

**Comment by mosra** [comment]
Magnum provides its own GL headers, so you're getting a conflict with the system-wide one. I wasn't aware of there being a `GLES3/gl32.h`, just pushed a fix for that in 80a844c6c5256b1c77a04db22f8042891b521c0b. Can you pull and try again?

Note that you should include Magnum GL headers *before* the system ones to avoid errors with missing definitions -- the system ones will never be as complete as Magnum needs, which is why it has its own.

**Comment by mgood7123** [comment]
a fix is to replace the egl and gl32 headers with Magnum/GL headers instead
